### ISSUE TITLE
Rewrite the `withResizeObserver` mixin as a reactive controller

### ISSUE THREAD

**Issue by hsablonniere** [CleverCloud/clever-components#551]
(title an description edited to reflect the latest proposition)

## Context

* We created the `withResizeObserver` mixin to hide the boilerplate of using `ResizeObserver` and provide nicers APIs for our components who need to be responsive
* Lit introduced the concept of [reactive controllers](https://lit.dev/docs/api/controllers/#ReactiveController)
  * This is clearly a better approach/solution than mixins
* The lit-labs project provides a [ready to use reactive controller](https://github.com/lit/lit/blob/main/packages/labs/observers/src/resize-controller.ts) to use the `ResizeObserver` API in a component

## About the [`ResizeController`](https://github.com/lit/lit/blob/main/packages/labs/observers/src/resize-controller.ts) from lit-labs

🙂 Pros:

* It doesn't do much things we don't need
* It seems very smart (just one `ResizeObserver` for many instances of the same component but multiple `observe` calls)

🤬 Cons:

* It doesn't provide the nice CSS API (with the automatic attributes on breakpoints) we have in our mixin
  * Most of those could be replaced by [CSS container queries](https://caniuse.com/css-container-queries) but the support is not good enough
* It just provides the `ResizeObserverEntry[]` but as we experienced it, it's not enough for us

## Proposition

* Create our own reactive controller reusing the code from our mixin
  * Try to see if we can reuse lit-labs's `ResizeController` or if we can get inspiration from it
* Replace all our mixin usage with the reactive controller
* Remove the mixin (is it a breaking change if it was not exposed / documented for public usage?)

WDYT?

**Comment by florian-sanders-cc** [comment]
Tested it very quickly to see if it also triggered the `ResizeObserver loop limit exceeded` like our `ResizeObserver` in the the `cc-invoice-table` component.

It does trigger the same error. 

The API feels easy to use so I still think it's interesting :+1:

**Comment by hsablonniere** [comment]
> It does trigger the same error.

@florian-sanders-cc Yes, it does not. I wrote about the loop errors in issue #808.

**Comment by pdesoyres-cc** [comment]
This is what we use today:

- resize observer mixin is used in 10 components (`cc-invoice-list`, `cc-invoice-table`, `cc-map`, `cc-orga-member-card`, `cc-overview`, `cc-pricing-product`, `cc-pricing-product-consumption`, `cc-tile-metrics`, `cc-tile-requests`, `cc-zone-input`)
- breakpoints feature is used in 4 components (`cc-invoice-list`, `cc-orga-member-card`, `cc-overview`, `cc-zone-input`)

**Comment by florian-sanders-cc** [comment]
@hsablonniere I've started working on this issue and I got a reactive controller working but I checked the Lit implementation to see if my controller was missing stuff implemented in their version.

Checking their code made me realize that we can actually use the lit-labs version: the controller accepts a callback.

The callback is called with two arguments: `entries` and `observer`.

Inside `entries`, you have a `target` (you can decide if you want to observe the host or an element inside it but in our use cases, the target will always be the `host`).

Since we have access to `target`, we can:

- use `getBoundingClientRect` like we currently do,
- get its publicly declared `breakpoints`,
- set attributes.

TLDR; I think I'll submit two PRs so we can decide if we go with our own controller or the lit-labs implementation.
